searchTree/searchTree.c

```diff
@@ -0,0 +1,151 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "searchTree.h"
+
+typedef struct Node {
+    NodeValue value;
+    Node* leftChild;
+    Node* rightChild;
+} Node;
+
+Node* createTreeNode(NodeValue value, int* errorCode) {
+    Node* newNode = calloc(1, sizeof(Node));
+    if (newNode == NULL) {
+        *errorCode = 1;
+        return NULL;
+    }
+    newNode->value = value;
+    return newNode;
+}
+
+void attachLeftChild(Node* node, Node* child) {
+    node->leftChild = child;
+}
+
+void attachRightChild(Node* node, Node* child) {
+    node->rightChild = child;
+}
+
+void insertNode(Node* node, NodeValue value, int* error) {
+    if (node->value.key == value.key) {
+        char* oldValue = node->value.value;
+        node->value = value;
+        free(oldValue);

```

Comment:
```suggestion
        free(node->value.value);
        node->value = value;
```
Вам же oldValue низачем, можно сразу удалить и присвоить. Единственное что тут тоже не присвоить, а скопировать, а копирование требует выделения памяти, а выделение памяти может закончиться с ошибкой, а мы не хотим модифицировать исходное дерево до того, как будем уверены, что сможем без ошибки выполнить операцию...